cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScaler.java

```diff
@@ -239,35 +235,59 @@ public void close() {
     /**
      * Closes the Zookeeper connection
      */
-    private void closeConnection(ZooKeeperAdmin zkAdmin) {
-        if (zkAdmin != null)    {
-            try {
-                zkAdmin.close((int) operationTimeoutMs);
-            } catch (Exception e) {
-                log.debug("Failed to close the ZooKeeperAdmin", e);
-            }
+    private Future<Void> closeConnection(ZooKeeperAdmin zkAdmin) {
+        Promise<Void> closePromise = Promise.promise();
+
+        if (zkAdmin != null) {
+            vertx.executeBlocking(promise -> {
+                try {
+                    zkAdmin.close((int) operationTimeoutMs);
+                    closePromise.complete();
```

Comment:
You should be calling `promise.complete` here. By calling `closePromise` you end up running the completion callback for `closePromise` on the worker pool thread, rather than the context thread.